JavaScript/es6Guidelines.md

```diff
@@ -0,0 +1,204 @@
+
+### Default Exports
+
+**Always add export default**
+
+When you don't know which property you should export as default a one think about the property which is the best representative of particular module or which one is going to be used externally most often.
+
+### Classes VS Modules
+
+**Favor Modules over Classes.**
+
+However, the question remains, when I should use class?  
+When it comes to answer to this question I'm too biased to answer that's why I recommend this two articles:
+
+*   [i-would-never-argue-that-es6-classes-are-clearly-a-better-choice-than-composition-modules-or-prototypal-OO](https://medium.com/@rauschma/i-would-never-argue-that-es6-classes-are-clearly-a-better-choice-than-composition-modules-or-891e462da85b#.bfhfjc74j)
+*   [a-simple-challenge-to-classical-inheritance-fans](https://medium.com/javascript-scene/a-simple-challenge-to-classical-inheritance-fans-e78c2cf5eead#.vh341a5iq)[](https://medium.com/javascript-scene/a-simple-challenge-to-classical-inheritance-fans-e78c2cf5eead#.vh341a5iq)
+
+### Imports first
+
+**Imports should be the first thing you put in your files.**  
+
+_Note: ES6 imports are declarative and meant for static analysis. They cannot be dynamic._  
+
+## ESLint
+
+### How to read .eslint.rc
+
+Here is small piece of our .eslint.rc file:
+```json
+"no-process-exit": 0,
+"no-restricted-modules": 0,
+"no-sync": 0,
+"array-bracket-spacing": [2, "never"]
+```
+What all those things mean?
+
+*   the key is the name of the rule - `"no-process-exit"`
+*   the number after key means:  
+
+    *   0 - turn the rule off
+    *   1 - turn the rule on as a warning (doesn't affect exit code)
+    *   2 - turn the rule on as an error (exit code is 1 when triggered)
+*   `[2, "never"]` - rule is turned on as an error and plugin called `never` is on 
+
+### Cascading and Hierarchy
+
+The configuration cascade works by using the closest `.eslintrc` file to the file being linted as the highest priority, then any configuration files in the parent directory, and so on.
+
+### Defined globals
+
+*   / - [/.eslintrc#L11](https://github.com/Wikia/mercury/blob/dev/.eslintrc#L11)
+*   /front - [/front/.eslintrc#L7](https://github.com/Wikia/mercury/blob/dev/front/.eslintrc#L7)
+*   /server/ - [/server/.eslintrc](https://github.com/Wikia/mercury/blob/dev/server/.eslintrc)
+
+### Most important enabled rules
+
+*   **Prefer object shorthand**
+
+    Example:
+
+    ```javascript
+    let bar = 'bar',
+        foo = 'foo';
+    return {
+      foo,
+      bar,
+      fizz: 'fizz'
+    };
+    ```
+
+*   **Prefer dot notation in objects**
+
+    Example:
+
+    ```javascript
+    point.x = 7;
+    ```
+
+*   **New line after variables declaration**
+
+    Example:
+
+    ```javascript
+    sum() {
+      const point = {
+        x: 5,
+        y: 7
+      };
+
+      //...rest of method's body
+    }
+    ```
+
+*   **Use `()=>` when possible**
+*   **Only one `var/let/const` declaration inside block**

```

Comment:
It's the same thing as https://github.com/Wikia/guidelines/blob/es6/JavaScript/es6Guidelines.md#const-and-let-over-var, maybe we should link it here?